test/app-luatest/gh_8196_crash_box_internal_compact_test.lua

```diff
@@ -0,0 +1,25 @@
+local t = require('luatest')
+local net = require('net.box')
+local server = require('luatest.server')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.server = server:new({alias = 'master'})
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+g.test_gh_8196 = function(cg)
+    local c = net.connect(cg.server.net_box_uri)
+    t.assert_equals(c.error, nil)
+    c:close()
```

Comment:
There's no need to connect to the server with net.box. `server.exec` user net.box under the hood so if the server fails it'll raise an error, anyway.